src/client_creds.rs

```diff
@@ -0,0 +1,114 @@
+///
+/// Note: This flow does not include authorization and therefore cannot be used
+/// to access or to manage the endpoints related to user private data in
+/// [`OAuthClient`](crate::endpoints::OAuthClient).
+///
+/// [reference]: https://developer.spotify.com/documentation/general/guides/authorization-guide/#client-credentials-flow
+/// [example-main]: https://github.com/ramsayleung/rspotify/blob/master/examples/client_creds.rs
+#[derive(Clone, Debug, Default)]
+pub struct ClientCredentialsSpotify {
+    pub config: Config,
+    pub creds: Credentials,
+    pub token: Option<Token>,
+    pub(in crate) http: HttpClient,
+}
+
+/// This client has access to the base methods.
+impl BaseClient for ClientCredentialsSpotify {
+    fn get_http(&self) -> &HttpClient {
+        &self.http
+    }
+
+    fn get_token(&self) -> Option<&Token> {
+        self.token.as_ref()
+    }
+
+    fn get_token_mut(&mut self) -> Option<&mut Token> {
+        self.token.as_mut()
+    }
+
+    fn get_creds(&self) -> &Credentials {
+        &self.creds
+    }
+
+    fn get_config(&self) -> &Config {
+        &self.config
+    }
+}
+
+impl ClientCredentialsSpotify {
+    /// Builds a new [`ClientCredentialsSpotify`] given a pair of client
+    /// credentials and OAuth information.
+    pub fn new(creds: Credentials) -> Self {
+        ClientCredentialsSpotify {
+            creds,
+            ..Default::default()
+        }
+    }
+
+    /// Build a new [`ClientCredentialsSpotify`] from an already generated
+    /// token. Note that once the token expires this will fail to make requests,
+    /// as the client credentials aren't known.
+    pub fn from_token(token: Token) -> Self {
+        ClientCredentialsSpotify {
+            token: Some(token),
+            ..Default::default()
+        }
+    }
+
+    /// Same as [`Self::new`] but with an extra parameter to configure the
+    /// client.
+    pub fn with_config(creds: Credentials, config: Config) -> Self {
+        ClientCredentialsSpotify {
+            config,
+            creds,
+            ..Default::default()
+        }
+    }
+
+    /// Tries to read the cache file's token, which may not exist.
+    ///
+    /// Similarly to [`Self::write_token_cache`], this will already check if the
+    /// cached token is enabled and return `None` in case it isn't.
+    #[maybe_async]
+    pub async fn read_token_cache(&self) -> Option<Token> {
+        if !self.get_config().token_cached {
+            return None;
+        }
+
+        let token = Token::from_cache(&self.get_config().cache_path)?;
+        if token.is_expired() {
+            // Invalid token, since it doesn't have at least the currently
+            // required scopes or it's expired.
+            None
+        } else {
+            Some(token)
+        }
+    }
+
+    /// Obtains the client access token for the app. The resulting token will be
+    /// saved internally.
+    #[maybe_async]
```

Comment:
Yes. This function calls `write_token_cache` at the end so it will be written (only if cached tokens are enabled, as you may see in said function's definition)